syntax="proto3";

message NodeDetails {
    string ip = 1;
    string port = 2;
    string information = 3;
}

message CLOutput{
    string output = 1;
}

message CLInput {
    string input = 1;
}

message Empty {

}

message AllNodes {
    repeated NodeDetails nodedetails = 1;
}

service CommandLine{
    rpc registerNode(NodeDetails) returns (Empty) {}
    rpc ping(Empty) returns(Empty) {}
    rpc getNodes(Empty) returns(AllNodes) {}
    rpc runTerminal(stream CLInput) returns(stream CLOutput) {}
}

service NodeConnection{
    rpc ping(Empty) returns (Empty) {}
}
